p2p/pex/pex_reactor.go

```diff
@@ -409,11 +409,15 @@ func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) {
 		// TODO: detect more "bad peer" scenarios
 		if _, ok := err.(p2p.ErrSwitchAuthenticationFailure); ok {
 			r.book.MarkBad(addr)
+			r.attemptsToDial.Delete(addr.DialString())
 		} else {
 			r.book.MarkAttempt(addr)
+			// FIXME: if the addr is going to be removed from the addrbook (hard to
```

Comment:
@ebuchman what do you think about this?

Reply:
Oooh. Good catch.

So the problem is that the addr might get removed from the AddrBook later once it's full, but it's still being tracked in attemptsToDial, and then will never get deleted ?

So our attemptsToDial map could get bloated with these bad peers.

I think these means we really do nee to unify the MarkBad mechanisms more intelligently.Maybe the AddrBook just needs access to the attemptsToDial somehow? 

Let's open a new issue for it. It's not critical right now but it is important.